jwk.go

```diff
@@ -719,3 +756,42 @@ func (key rawJSONWebKey) symmetricKey() ([]byte, error) {
 	}
 	return key.K.bytes(), nil
 }
+
+// Remove duplicates from a slice
+func removeDups(elements []string) []string {
+	// Use map to record duplicates as we find them.
+	encountered := map[string]bool{}
+	result := []string{}
+
+	for v := range elements {
+		if encountered[elements[v]] {
+			// Do not add duplicate.
+		} else {
+			encountered[elements[v]] = true
+			result = append(result, elements[v])
+		}
```

Comment:
Isn't the if/else here redundant? You can just shove everything into the keys of the map, and the map will dedup.